libs/framework/include/celix/BundleActivator.h

```diff
@@ -84,9 +82,9 @@ namespace celix {
         celix_status_t destroyActivator(void *userData) {
             auto *data = static_cast<BundleActivatorData<I> *>(userData);
             std::weak_ptr<celix::BundleContext> ctx = data->ctx;
-            std::weak_ptr<celix::dm::DependencyManager> dm = data->dm;
+            std::weak_ptr<celix::dm::DependencyManager> dm = data->ctx->getDependencyManager();
             auto bndId = data->bndId;
-            data->dm->clear();
+            data->ctx->getDependencyManager()->clear();
```

Comment:
Nitpick: `dm->clear();` would be clearer here

Reply:
note that the clear is done before the `delete data`, so I think a lock() on a weak ptr is not needed and prefer using the data->ctx->getDependencyManager().

I moved the statement a bit higher, hopefully that is more clearer.